src/centralserver/admin-ui/src/views/Settings/SystemSettings/SystemSettings.vue

```diff
@@ -378,29 +380,32 @@ export default (
       ];
     },
 
-    serverAddress(): string {
-      return this.$store.getters[StoreTypes.getters.SYSTEM_STATUS]
-        ?.initialization_status?.central_server_address;
+    serverAddress(): string | undefined {
+      return this.getSystemStatus?.initialization_status
+        ?.central_server_address;
     },
-    instanceIdentifier(): string {
-      return this.$store.getters[StoreTypes.getters.SYSTEM_STATUS]
-        ?.initialization_status?.instance_identifier;
+    instanceIdentifier(): string | undefined {
+      return this.getSystemStatus?.initialization_status?.instance_identifier;
     },
   },
   methods: {
+    ...mapActions(notificationsStore, ['showError', 'showSuccess']),
+    ...mapActions(systemStore, [
+      'fetchSystemStatus',
+      'updateCentralServerAddress',
+    ]),
     async onServerAddressSave(serverAddress: string): Promise<void> {
       try {
-        await this.$store.dispatch(
-          StoreTypes.actions.UPDATE_CENTRAL_SERVER_ADDRESS,
-          {
-            central_server_address: serverAddress,
-          },
-        );
-        await this.$store.dispatch(StoreTypes.actions.FETCH_SYSTEM_STATUS);
-        await this.$store.dispatch(
-          StoreTypes.actions.SHOW_SUCCESS,
+        await this.updateCentralServerAddress({
+          central_server_address: serverAddress,
+        });
+
+        await this.fetchSystemStatus();
+
+        this.showSuccess(
```

Comment:
I think `await` should be added for this , too. Otherwise isEditingServerAddress can get false too early.

Reply:
`await` would need a promise to be returned. `showSuccess` doesn't return anything because it's meant to be kind of "fire and forget" action.

Reply:
Ajoitus on tässä kai ainoa mahdollisesti virhettä aiheuttava asia. 
await on mahdollinen myös tässä: 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await
"If the value of the expression following the await operator is not a Promise, it's converted to a resolved Promise"

Reply:
I don't see any issue with timing success snackbar. It's fired to show, then it shows up and goes away on it's own and rest of the app can go on during that time. Even if the snackbar would fail for some reason it should not stop anything.